control.go

```
@@ -322,7 +322,11 @@ func (c *controlConn) setupConn(conn *Conn, sessionInit bool) error {
 	iter := conn.querySystemLocal(context.TODO())
 	host, err := c.session.hostInfoFromIter(iter, conn.host.connectAddress, conn.r.RemoteAddr().(*net.TCPAddr).Port)
 	if err != nil {
-		return err
+		iter.Close()
```

Should we handle the error returned by Close()?

---

This was mostly to just cleanup the iterator. We should probably return the error from `hostInfoFromIter`.